### PR TITLE
Flatpak build recipe added

### PR DESCRIPTION
**Pull request by sahaRatul**
```SUMMARY: Build "Adds Flatpak build recipe"```
# COMPILING GUIDE FOR LINUX (USING FLATPAK)

## PREREQUISITES

- Install `flatpak` and `flatpak-builder` from your distribution's package manager.
- Install `freedesktop-platform` & `freedesktop-sdk` using `flatpak install flathub org.freedesktop.Platform//18.08 org.freedesktop.Sdk//18.08`

## BUILD

- Clone the source code and run `flatpak-builder --repo=repo --ccache --force-clean build-dir org.Cataclysm.DDA.json` or `flatpak-builder --repo=repo --ccache --force-clean build-dir org.Cataclysm.DDA.Tiles.json` to build the ncurses or tiles version of the application.

## INSTALLATION

Run `flatpak --user install test-repo org.Cataclysm.DDA` (or) `flatpak --user install test-repo org.Cataclysm.DDA.Tiles` to install the game.
Note: If you are installing the game for first time you will have to set up local repo. Run this `flatpak --user remote-add --no-gpg-verify test-repo repo`

## STARTING THE GAME

`flatpak run org.Cataclysm.DDA` (or) `flatpak run org.Cataclysm.DDA.Tiles`
**or**
Launch the Game from Applications -> Games -> Cataclysm: Dark Days Ahead (or) Cataclysm: Dark Days Ahead - Tiles.
**Note** : For the ncurses version, terminal will just show a blank screen on launch. Just press any button to start the game. 

## PERMISSIONS

- **Ncurses version**
  - Home Directory
- **Tiles Version**
  - Home Directory
  - Wayland
  - Fallback-X11 - (incase Wayland doesn't work)
  - IPC - (Required for X11)
  - PulseAudio
  - DRI

## [Publishing on Flathub](https://github.com/flathub/flathub/wiki/App-Submission)

The sources section in build manifest should be modified as below
`"sources": [
        {
          "type": "git",
          "url": "https://github.com/CleverRaven/Cataclysm-DDA.git"
        }
      ]`
**or**
[Hosting in custom repo](http://docs.flatpak.org/en/latest/publishing.html)

### CODE REVIEW

**Comment by sahaRatul** [comment]
## Build
- Install flatpak and flatpak-builder from your distributions package manager.
- Install freedesktop sdk using `flatpak install flathub org.freedesktop.Platform//18.08 org.freedesktop.Sdk//18.08`
- Clone the source code.
- Run `flatpak-builder --repo=repo --force-clean build-dir org.Cataclysm.DDA.json` to build the application.

## Installation
Run `flatpak --user install test-repo org.Cataclysm.DDA` to install the game.
Note: If you are installing the game for first time you will have to set up local repo. Run this `flatpak --user remote-add --no-gpg-verify test-repo repo`

## Running the game
`flatpak run org.Cataclysm.DDA`
#### or
Launch the Game from Applications -> Games -> Cataclysm: Dark Days Ahead.
Note: The terminal will just show a blank screen on launch. Just press any button to start the game. 

## Publishing on Flathub
http://docs.flatpak.org/en/latest/repositories.html

**Comment by AMurkin** [comment]
Instructions probably should be commited in the .md file like others: [COMPILING-CMAKE.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/COMPILING-CMAKE.md), [COMPILING-MSYS.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/COMPILING-MSYS.md) etc.

**Comment by AMurkin** [comment]
> Note : For the ncurses version, terminal will just show a blank screen on launch. Just press any button to start the game.

Is it redundant now?

**Comment by sahaRatul** [comment]
> 
> 
> > Note : For the ncurses version, terminal will just show a blank screen on launch. Just press any button to start the game.
> 
> Is it redundant now?

Nope, this still occurs on ncurses version. I got rid of background terminal in the tiles version

File: COMPILING-FLATPAK.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# COMPILING GUIDE FOR LINUX (USING FLATPAK)
```
Comment (AMurkin) [inline COMPILING-FLATPAK.md]: I think sentence-cased headings will look better.

File: COMPILING-FLATPAK.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## PREREQUISITES
+
+- Install `flatpak` and `flatpak-builder` from your distribution's package manager.
+- Install `freedesktop-platform` & `freedesktop-sdk` using `flatpak install flathub org.freedesktop.Platform//18.08 org.freedesktop.Sdk//18.08`
+
+## BUILD
+
+- Clone the source code and run `flatpak-builder --repo=repo --ccache --force-clean build-dir org.Cataclysm.DDA.json` or `flatpak-builder --repo=repo --ccache --force-clean build-dir org.Cataclysm.DDA.Tiles.json` to build the ncurses or tiles version of the application.
```
Comment (AMurkin) [inline COMPILING-FLATPAK.md]: Here is only one item in the markdown list.

File: COMPILING-FLATPAK.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Run `flatpak --user install test-repo org.Cataclysm.DDA` (or) `flatpak --user install test-repo org.Cataclysm.DDA.Tiles` to install the game.
+Note: If you are installing the game for first time you will have to set up local repo. Run this `flatpak --user remote-add --no-gpg-verify test-repo repo`
+
+## STARTING THE GAME
+
+`flatpak run org.Cataclysm.DDA` (or) `flatpak run org.Cataclysm.DDA.Tiles`
+**or**
+Launch the Game from Applications -> Games -> Cataclysm: Dark Days Ahead (or) Cataclysm: Dark Days Ahead - Tiles.
+**Note** : For the ncurses version, terminal will just show a blank screen on launch. Just press any button to start the game.
```
Comment (AMurkin) [inline COMPILING-FLATPAK.md]:
```suggestion
**Note**: For the ncurses version, terminal will just show a blank screen on launch. Just press any button to start the game. 
```

File: COMPILING-FLATPAK.md (hunk 1 above)
Comment (sahaRatul) [reply COMPILING-FLATPAK.md]: You mean like this?

# Compiling Guide for Linux

**Comment by kevingranade** [comment]
The json linting script needs to be adjusted to exclude these new json files.

File: .gitignore [hunk 4]
```diff
@@ -170,3 +170,8 @@ Xcode/
 
 # gdb
 .gdb_history
+
+# Flatpak build artifacts
+/.flatpak-builder/
+/repo/
+/build-dir/
```
Comment (ZhilkinSerg) [inline .gitignore]:
```suggestion
/build-dir/

```

You need an empty line as last line of any file.

File: org.Cataclysm.DDA.Tiles.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      ],
+      "sources": [
+        {
+          "type": "dir",
+          "path": "."
+        }
+      ]
+    }
+  ]
+}
```
Comment (ZhilkinSerg) [inline org.Cataclysm.DDA.Tiles.json]:
```suggestion
}

```

File: org.Cataclysm.DDA.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      ],
+      "sources": [
+        {
+          "type": "dir",
+          "path": "."
+        }
+      ]
+    }
+  ]
+}
```
Comment (ZhilkinSerg) [inline org.Cataclysm.DDA.json]:
```suggestion
}

```

**Comment by ZhilkinSerg** [comment]
There is following unwanted output when building in MSYS:

```
find: paths must precede expression: org.Cataclysm.DDA.Tiles.json
Usage: find [-H] [-L] [-P] [-Olevel] [-D help|tree|search|stat|rates|opt|exec] [path...] [expression]
find: paths must precede expression: org.Cataclysm.DDA.Tiles.json
Usage: find [-H] [-L] [-P] [-Olevel] [-D help|tree|search|stat|rates|opt|exec] [path...] [expression]
```

I don't think it breaks anything, but I would need to look into suppressing it.

**Comment by sahaRatul** [comment]
> The json linting script needs to be adjusted to exclude these new json files.

You mean those should be added on json_blacklist file?

**Comment by ZhilkinSerg** [comment]
> 
> 
> There is following unwanted output when building in MSYS:
> 
> ```
> find: paths must precede expression: org.Cataclysm.DDA.Tiles.json
> Usage: find [-H] [-L] [-P] [-Olevel] [-D help|tree|search|stat|rates|opt|exec] [path...] [expression]
> find: paths must precede expression: org.Cataclysm.DDA.Tiles.json
> Usage: find [-H] [-L] [-P] [-Olevel] [-D help|tree|search|stat|rates|opt|exec] [path...] [expression]
> ```
> 
> I don't think it breaks anything, but I would need to look into suppressing it.

It seems this is not the fault of your PR - we have missing quotation marks in Makefile:

image

This change also hides from linter.

**Pull request merged**